doc: project_roles: define inactivity criteria for role retirement (#113563)
zephyrproject-rtos/zephyr: doc/project/project_roles.rst

Role Retirement
###############

> Individuals approved by the TSC or representatives of the TSC to fill a project

Reviewer: This first paragraph meant to cover more than maintainers. So it should not be juts removed. (And there is no need to remove it)

Author: I kept it in v2

individual’s GitHub user name from the relevant sections of the :ref:`maintainers_file`
in the Zephyr repository, and may be initiated by the TSC, representatives of
the TSC, or by the individuals themselves. Maintainers may also initiate the
removal of inactive collaborators in their area.

A maintainer or collaborator is considered inactive if, over a continuous period
of six months, they have had no involvement in the project. For the purposes of
this policy, involvement includes any of the following: merged or submitted code
contributions, pull request reviews, participation in issue or pull request
discussions, RFC authorship or feedback, or attendance in TSC or Working Group

> meetings relevant to their area.

Reviewer: This is not good.
a) 6 months is an eternity for areas with a lot of activity.
b) It is irrelevant if a person is active in other areas if they have abandoned another they are the maintainer of.

There is no need to clump together maintainers and collaborators in the same procedure. The consequences of an inactive collaborator are trivial in comparison, and therefore the procedure can be way more lax / left for the respective maintainer discretion.

Author: As discussed last week in the Process WG I've updated with 30 days of inactivity.

Inactivity is tracked by the TSC or its designated representatives. After three

> months of inactivity, a courtesy reminder may be sent to the individual to flag

Reviewer: 3 months would be too long for active areas to start pinging an absent maintainer.

Author: updated to 30 days

that they are approaching the inactivity threshold. After six months, a formal
notice is sent by email and posted to a public tracking issue, with the other
maintainers of the affected area copied where applicable. The notice asks whether
the individual is able to remain involved in the project, recognizing that an
absence may be due to temporary circumstances such as workload, illness, or

> personal leave.

Reviewer: a) 6 months is too long
b) pinging should have been done much earlier (in this PR it said above 3 months..)
c) Contacting somebody could happen thru any medium, not just email.

Author: a) s/6 months/30 days
b) I've added 30 days, but maybe it should be earlier? IMO it shouldn't. 
c) Email is the formal announcement. Someone can ping the inactive maintainer earlier than 30 days through any medium. But if there is no response, or no involvement even after pinging them on Discord, for example, we send the "formal" announcement

If no response is received within two weeks, the individual is removed from the

> :ref:`maintainers_file`. If the individual responds and wishes to remain, they are

Reviewer: This could be done just sending a PR to the maintainer's file directly and holding it/setting DNM on it.

Author: You proposed opening a PR to remove the maintainer and adding a DNM tag to it for two weeks or until the individual responds?
I believe that can be done.

Reviewer: Yes please, a bit less red tape.

associated with their role may be requested by the TSC to retire from that role.

Retirements of inactive maintainers or collaborators are reflected by removing the
individual's GitHub user name from the relevant sections of the :ref:`maintainers_file`
in the Zephyr repository, and may be initiated by the TSC, representatives of the
TSC, or by the individuals themselves. Maintainers may also initiate the removal of
inactive collaborators in their area.

A maintainer or collaborator is considered inactive if, over a continuous period of

> 30 days, they have performed none of the following activities within their area of

Reviewer: Major:
a) This definition is not good.
If an area has no pending activity, there should be no "running timer". (There is lots of areas where nothing needs to happen for 30 days)
Being inactive should be defined as not doing something which actually needs doing for a period of time.

b) This would at least cover people who is completely gone, which seems better than nothing;
But by being so specific, if we would attempt to remove a person who is not doing what they should, it could result in a lot of "lawyering".
Meaning, this definition of activity includes so many minor and fluffy things that a person could have practically abandoned their area, but by just doing some triviality every couple of weeks they would not qualify. We do need to have a definition that would cover a maintainer failing to do what is expected at a much more reasonable level.
Also this conditions would be imposible to automate.

Author: updated to link to the responsibilities already defined in this file.

responsibility:
- authored, merged, or landed a pull request;
- submitted a review or a substantive comment on a pull request;
- triaged, responded to, or closed an issue;
- authored or provided feedback on an RFC or design proposal;

> - attended a TSC, Working Group, or a meeting relevant to their area;

Reviewer 2: WG visitors are not bound to GitHub names. 
It's better to delete requirements that are not tracked automatically by GH.

- participated in a discussion on the mailing list, Discord, or GitHub Discussions
concerning their area.

> Activity outside an individual's area of responsibility does not by itself satisfy

Reviewer: ```suggestion
Activity outside that maintainer area of responsibility does not by itself satisfy
```

Author: updated

> this requirement. Automated or trivial contributions, such as whitespace fixes
> or bot-assisted updates, are not counted.

Reviewer: Minor:
```suggestion
this requirement.
```
I do not see why we'd mention this. It is not a case I have seen or would ever expect.

Author: removed.
